### PR TITLE
(rpc_server): fix transaction details borsh deserialize

### PR DESCRIPTION
**Pull request by kobayurii**
Since https://github.com/near/nearcore/pull/10676
the ReceiptEnumView has been changed to include `is_promise_yield` field
which is not present in the old version of the ReceiptEnumView
by deserializing the old version of the ReceiptEnumView and converting it to the new version

solved: https://github.com/near/read-rpc/issues/224

### CODE REVIEW

**Comment by kobayurii** [comment]
@race-of-sloths please, include my PR in the Race

**Comment by race-of-sloths** [comment]
@kobayurii Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
Do you want to apply for monthly streak? Get 8+ score for a single PR this month and receive boost for race-of-sloths!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/kobayurii?type=bot&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/kobayurii?type=bot&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/kobayurii)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command. Alternatively, autoscoring [1,2] will be applied for this pull request

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/kobayurii)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

File: rpc-server/src/modules/transactions/mod.rs
Comment (khorolets) [inline rpc-server/src/modules/transactions/mod.rs:2]: i don't think this change was intentional

**Review by khorolets — APPROVED** [review]
Okay, pretty weird though, but does the job. Thank you!

**Comment by race-of-sloths** [comment]
🥁  __Score it!__

@khorolets, please score the PR with `@race-of-sloths score [1/2/3/5/8/13]`. The contributor deserves it.
If no scoring is provided within 24 hours, this PR will be scored as 1 🦥

**Comment by khorolets** [comment]
@race-of-sloths score 3

**Pull request merged**
